specification/basic_concepts/basic_concepts.tex

```diff
@@ -29,38 +29,30 @@ \chapter{Basic concepts}\label{sec:basic_concepts}
 A DSDL description can be used to automatically generate the serialization/deserialization code
 for every defined data structure in a particular programming language.
 DSDL ensures that the worst case memory footprint and computational complexity per data type
-are constant and easily predictable, which is paramount for hard real-time and safety-critical applications.
+are constant and easily predictable.
 
 On top of the standard data types, UAVCAN defines a set of standard high-level functions including:
-node health monitoring, network discovery, time synchronization, firmware update,
+node health monitoring, node discovery, time synchronization, firmware update,
 plug-and-play node support, and more.
 For more information see chapter \ref{sec:application_layer}.
 
 Serialized message and service data structures are exchanged by means of the transport
 layer (chapter \ref{sec:transport_layer}),
-which implements automatic decomposition/reassembly of long transfers into/from several transport
+which implements automatic decomposition/reassembly of long transfers from/into several transport
```

Comment:
> which implements automatic decomposition of long transfers into several transport frames

> which implements automatic reassembly of long transfers from several transport frames

Decomposition from; reassembly into. That seemed correct to me, although I could be missing something.

Reply:
We're using / here in place of 'or' so I was trying to order the sentence so decomposition would align with from and reassembly would align with into but now that I read it I think this is backwards. That sais, this sentence would make English majors riot so perhaps we should just avoid trickery and go with:

```
Serialized message and service data structures are exchanged by means of the transport
layer (chapter \ref{sec:transport_layer}), which implements automatic decomposition-into and reassembly-from long transfers to/from several transport frames\footnote{Here and elsewhere, a \emph{transport frame} means a block of data
that can be atomically exchanged over the network, e.g., a CAN frame.},
allowing nodes to exchange data structures of arbitrary size.
```